api/FetchClient.js

```diff
@@ -0,0 +1,37 @@
+const fetch = require('node-fetch');
+const isTokenValid = require('../utils/isTokenValid');
+
+const FetchClient = class {
+    constructor({ url, method, headers, body }) {
+        const defaultHeaders = {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+        };
+        this.url = url || '';
+        this.method = method || 'GET';
```

Comment:
Nice! It's also possible to set defaults in the function destructuring itself like this

```js
    constructor({ url = '', method = 'GET', headers, body } = {}) {
        const defaultHeaders = {
            Accept: 'application/json',
            'Content-Type': 'application/json',
        };
        this.url = url;
        this.method = method;
        this.headers = { ...defaultHeaders, ...headers };
        this.body = body;
    }
```

I'm also setting a default empty object to the argument if no argument object was provided. This makes it so that the class can be called with `new FetchClient()` without the need for a config, but it doesn't make much sense in this case since you probably always want, at least the url passed in.

This comment is just a note, we don't need to change anything